File: internal/commands/rename.go
package commands

import "minicode/internal/storage"

func registerRename(r *Registry) {
	r.Register(&Command{Name: "rename", Description: "Rename current session", Kind: Handler,
		Handler: func(args []string, ctx Context) (Result, error) {
			newName := ""
			if len(args) > 0 {
				newName = args[0]
			}
			if newName != "" && ctx.Agent != nil {
				oldName := ctx.Agent.SessionName()
				_ = storage.NewSessionManager().Rename(oldName, newName)
				ctx.Agent.SetSession(newName)
			}
			return HandledResult{}, nil
		},
	})
}

File: src/ui/commands/builtin/rename.ts
import type { CommandHandler } from "../registry.js";

export const renameCommand: CommandHandler = {
  name: "rename",
  description: "Rename current session",
  handler: async (args, ctx): Promise<void> => {
    const newName = args.join(" ");
    if (newName) {
      await ctx.renameCurrentSession(newName);
    }
  },
};
